### PR TITLE
[HAR-13]Bot tells you your reserved total, if you have reserves

### PR DESCRIPTION
**Pull request by Veetiknen**
Also changed the user finding a bit. The columns were previously hard coded and now it looks for the Headers first.

### CODE REVIEW

File: src/functions/utils/messages.ts [hunk 1]
```diff
@@ -2,6 +2,8 @@
 export const messages = {
   bonusMessage:
     "Your total usable bonus is: {remainingTotalBudget}€\nAmount to be paid as salary: {remainingTotalPayable}€",
+  reservedBonusMessage:
```
Comment (lockmhe) [inline src/functions/utils/messages.ts]: Do not duplicate texts. Have the sentence about reserved amount in a separate string and then combine on the calling side when necessary.

File: src/functions/utils/slackBot.ts [hunk 2]
```diff
@@ -31,23 +31,33 @@ export class SlackBot {
...
-          messages.bonusMessage
-            .replace("{remainingTotalBudget}", data.usableBonus.toString())
-            .replace("{remainingTotalPayable}", data.payableBonus.toString())
-        );
+        if (data.reservedTotal > 0) {
+          await respond(
+            messages.reservedBonusMessage
+              .replace("{remainingTotalBudget}", data.usableBonus.toFixed(2))
+              .replace("{remainingTotalPayable}", data.payableBonus.toFixed(2))
+              .replace("{reservedTotal}", data.reservedTotal.toFixed(2))
```
Comment (lockmhe) [inline src/functions/utils/slackBot.ts]: Combine from 2 different strings here (see earlier comment) about not duplicating loc texts

File: src/functions/utils/userRowFinder.ts [hunk 3]
```diff
@@ -5,10 +5,39 @@ import { BonusAmounts, ExcelData } from "./types";
 export function findUserRow(excelData: ExcelData, userName: string): BonusAmounts {
   const rows = excelData.values;
 
-  for (let i = 3; i < rows.length; i++) {
+  if (rows.length === 0) {
```
Comment (lockmhe) [inline src/functions/utils/userRowFinder.ts:8]: Was the idea about new page where the columns are linked dropped?

**Review by lockmhe — CHANGES_REQUESTED** [review]

File: src/functions/utils/userRowFinder.ts (hunk 3 above)
Comment (Veetiknen) [reply src/functions/utils/userRowFinder.ts:8]: Yes that was dropped.

**Review by lockmhe — APPROVED** [review]

**Pull request merged**